File: backend/services/error_reporting/fingerprint.py
import hashlib


def compute_fingerprint(source, error_type, request_path, status_code, kind):
    raw = "|".join([
        str(source or ""),
        str(error_type or ""),
        str(request_path or ""),
        str(status_code or ""),
        str(kind or ""),
    ])
    return hashlib.sha256(raw.encode()).hexdigest()[:64]

File: mobile/lib/core/error_reporting/fingerprint.dart
import 'dart:convert';
import 'package:crypto/crypto.dart';

String computeFingerprint({
  required String errorType,
  String? requestPath,
  int? statusCode,
}) {
  final input = '$errorType|${requestPath ?? ''}|${statusCode ?? ''}';
  final digest = sha256.convert(utf8.encode(input));
  return digest.toString().substring(0, 16);
}
